modin/experimental/core/execution/native/implementations/hdk_on_native/calcite_builder.py

```diff
@@ -108,6 +109,51 @@ def gen_reduce_expr(self):
             """
             pass
 
+    class CompoundAggregate2(CompoundAggregate):
+        """
+        A base class for a compound aggregate that require 2 input arguments.
+
+        The first argument must be an `InputRefExpr` and the second one - `LiteralExpr`.
+
+        Parameters
+        ----------
+        agg : str
+            Aggregate name.
+        builder : CalciteBuilder
+            A builder to use for translation.
+        arg : List of BaseExpr
+            Aggregate arguments.
+        dtype : dtype, optional
+            Aggregate data type. If not specified, `_dtype` from the first argument is used.
+        """
+
+        def __init__(self, agg, builder, arg, dtype=None):
+            assert isinstance(arg[0], InputRefExpr)
+            assert isinstance(arg[1], LiteralExpr)
+            super().__init__(builder, arg)
```

Comment:
so we pass 3 arguments here for a quantile, is this intended that the 3rd argument is ignored here?

Reply:
It's not ignored, it's handled here - https://github.com/modin-project/modin/pull/6528/files#diff-7972fd5dddcdf656c95eae334851ace130660d9aff5751510b11e6b62fd82187R404 . The quantile aggregation requires 3 arguments. The second one is added as a table column (the same as for TopK), and the 3-rd one (interpolation) is added as a new json attribute.